doc/schema-migration.md

```diff
@@ -0,0 +1,52 @@
+# Schema Migration or Norms
+
+Schema migration with Datahike is the evolution of your current schema into a future schema.
+We are calling the schema migrations 'norms' to avoid misunderstandings with a migration
+from an older version of Datahike to a newer version.
+
+When we are speaking of changes to your schema, these should always add new definitions and
+never change existing definitions. In case you want to change existing data to a new format
+you will have to create a new schema and transact your existing data transformed again. A
+good intro to this topic [can be found here](https://docs.datomic.com/cloud/schema/schema-change.html).
+
+## How to migrate
+
+1. Create a folder of your choice, for now let's call it `migrations`. In this folder you
+create a new file with an edn-extension like `001-my-first-norm.edn`. Preferably you name the
+file beginning with a number. Please be aware that the name of your file will be the id of
+your norm. Taking into account that you might create some more norms in the future
+you should left-pad the names with zeros to keep a proper sorting. Keep in mind that your
+norms are transacted sorted after your chosen ids one after another. Spaces will be replaced
+with dashes to compose the id.
+
+2. Write the transactions itself into your newly created file. The content of the file needs
+to be an edn-map with one or both of the keys `:tx-data` and `tx-fn`. `:tx-data` is just
+transaction data in the form of a vector, `:tx-fn` is a function that you can run during the
+execution to migrate data from one attribute to another for example. This function needs to
+be qualified and callable from the classpath. It will be evaluated during the migration and
+needs to return transactions. These transactions will be transacted with `:tx-data` together
```

Comment:
Does this mean the transaction data needs to fully fit into memory? This can easily become a problem if the database is getting big. We can probably ignore this for now, but should at least make this constraint explicit.

Reply:
right, that's an issue and we would have to use batching I guess.

Reply:
Yes. This might be non-trivial to add right now, but it will put a limit on the size of databases you can migrate, which is not good, as you can run into dead ends with not being able to run the migration scripts any more once the database gets too large.